docs: "What's new in Svelte" August newsletter (#2116)
sveltejs/svelte.dev: apps/svelte.dev/content/blog/2026-08-01-whats-new-in-svelte-august-2026.md

---
title: "What's new in Svelte: August 2026"
description: 'The SvelteKit 3 preview lands with new $app modules, zero-config error props and refreshAll'
author: Dani Sandoval
authorURL: https://dreamindani.com
---

> The biggest news this month is the first `@next` releases of SvelteKit 3. Eight preview builds shipped in July: previewing new `$app/manifest` and `$app/service-worker` modules, `refreshAll` in place of `invalidateAll`, tracing out of the experimental namespace and a lot more. It is prerelease, but it's worth trying out to see what's coming to SvelteKit!

Reviewer: ```suggestion
The biggest news this month is the first `@next` releases of SvelteKit 3. Twelve preview versions shipped in July: previewing new `$app/manifest` and `$app/service-worker` modules, improved API availability and type checking in service workers, tracing out of the experimental namespace and a lot more. It's a prerelease, but it's worth trying out to see what's coming to SvelteKit!
```

---
title: "What's new in Svelte: August 2026"
description: 'The SvelteKit 3 preview lands with new $app modules, zero-config error props and refreshAll'
author: Dani Sandoval
authorURL: https://dreamindani.com
---

The biggest news this month is the first `@next` releases of SvelteKit 3. Thirteen preview versions shipped in July: previewing new `$app/manifest` and `$app/service-worker` modules, improved API availability and type checking in service workers, tracing out of the experimental namespace, shallow routing baked into `goto` and a lot more. It's a prerelease, but it's worth trying out to see what's coming to SvelteKit!

Alongside the preview releases, the stable line kept moving with `submitted` on remote forms and a new home for `defineEnvVars`. The language tools also picked up zero-config `+error.svelte` props so error pages get their `page` and `error` types with no extra setup.

> And, in case you missed it, [Svelte Summit Ljubljana 2026](https://www.reddit.com/r/sveltejs/comments/1v8bdel/svelte_summit_ljubljana_2026_announcement/) is happening November 19-20, with a workshop day followed by a conference day. Save the date!

Reviewer: ```suggestion
And, in case you missed it, [Svelte Summit Ljubljana 2026](https://sveltesummit.com) is happening November 19-20, with a workshop day on November 18, the day before the summit. Save the date!
```
Link straight to the summit site and clears up that 19-20 is the summit while 18 is the workshop

## What's new in SvelteKit

- Remote forms now expose a `submitted` property so you can react to the moment a form is submitted without waiting for the response (**2.69.0**, [Docs](https://svelte.dev/docs/kit/remote-functions#form), [#14811](https://github.com/sveltejs/kit/pull/14811))
- `defineEnvVars` has moved from `@sveltejs/kit` to `@sveltejs/kit/env` so environment helpers live in a dedicated subpath (**2.70.0**, [Docs](https://svelte.dev/docs/kit/@sveltejs-kit-env#defineEnvVars), [#16378](https://github.com/sveltejs/kit/pull/16378))

> - CSRF protection now stays on in production builds even when `NODE_ENV` is set to something other than `production` (**2.70.0**, [Docs](https://svelte.dev/docs/kit/configuration#csrf), [#16313](https://github.com/sveltejs/kit/pull/16313))

Reviewer: ```suggestion
```
I wouldn't bother highlighting this bug fix. It also links to the docs of an option which is deprecated.

---

## Community Showcase

### Apps & Sites built with Svelte

- [Recipe Jar](https://recipejar.app) is a local-first recipe keeper built as a PWA with runes, IndexedDB and no backend ([GitHub](https://github.com/sbmagar13/recipe-jar))
- [Doota](https://doota.dev) reimagines email as a chat interface, built entirely on Svelte and Cloudflare's edge stack ([GitHub](https://github.com/etherCorps/doota))
- [Loot Raiders](https://www.reddit.com/r/sveltejs/comments/1uv3mhx/i_built_an_inventory_extraction_game_with_svelte_5/) is a browser-based inventory extraction game built with SvelteKit, GSAP and Howler
- [Motioner](https://motioner.app) is a browser-based Figma-style design tool where every frame doubles as an animation timeline
- [The Prototype](https://theprototype.app) is a collaborative 3D scene editor built on Threlte, with peer-to-peer sync over WebRTC and a full runes-based state system
- [clocks.dev](https://clocks.dev) is a community-built gallery where every clock is a Svelte component and anyone can contribute a new design

> - [Flatxel](https://flatxel.com) is an r/place-style collaborative pixel canvas built with SvelteKit

Reviewer: ```suggestion
- [Flatxel](https://flatxel.com) is an r/place-style collaborative pixel canvas
```
Most of the websites listed are built with SvelteKit but only two of them mention it. I wonder if we should just do away with the wording or add it to all which use it

Author: Good call. I often copy/paste from the apps' github description to get the first draft of the sentence. I CMD+F all mentions of Svelte but I forgot to do SvelteKit

- [Gabble](https://gabble.gg) is a daily word game PWA that mixes Wordle and Boggle mechanics
- [Can you terraform Mars?](https://www.nature.com/immersive/d41586-026-01978-8/index.html) is an interactive Nature featured article

### Learning Resources

_Featuring Svelte Contributors and Ambassadors_

- [atproto x npmx x svelte meetup](https://www.youtube.com/live/z3U3yUWrml4?si=BeV9eTgVyl0HghJv) happened in Berlin after Local-First Conf - a crossover satellite event with the AT Protocol, npmx, and Svelte communities!
- [Agentic Engineering with Svelte](https://mainmatter.com/blog/2026/07/28/agentic-engineering-with-svelte/) by Paolo Ricciuti walks through how Mainmatter is building agent-driven workflows on top of Svelte and SvelteKit

_This Week in Svelte_

- [Ep. 145](https://www.youtube.com/watch?v=1QJ5dFvbgrI) - Changelog, Wuchale
- [Ep. 146](https://www.youtube.com/watch?v=EjpIXG9dGtA) - Changelog, Chatto
- [Ep. 147](https://www.youtube.com/watch?v=SfZx_D0bI7E) - Changelog

_To Read_

> - [Migrated a dashboard application from React to Svelte](https://www.reddit.com/r/sveltejs/comments/1uqy6ub/migrated_a_dashboard_application_from_react_to/) - a detailed report from an engineer who cut their bundle size roughly in half after moving from TanStack Start + React Query to SvelteKit and remote functions

Reviewer: ```suggestion
- [Migrated a dashboard application from React to Svelte](https://www.reddit.com/r/sveltejs/comments/1uqy6ub/migrated_a_dashboard_application_from_react_to/) - an account from an engineer who cut their bundle size roughly in half after moving from TanStack Start + React Query to SvelteKit and remote functions
```
Changing the wording to temper expectations a little. A lot of these refactoring reads aren't usually very detailed about the process and learnings. I think there was only one done by a company pre-LLM era that shared a lot of feedback on what they encountered

> - [I built a Rust toolchain for Svelte](https://blog.baseballyama.com/posts/20260721-rsvelte) by Yamagishi Kazutoshi walks through `rsvelte`, a Rust reimplementation of the Svelte compiler, formatter, linter, type checker and LSP tested against 12,000+ real components

Reviewer: ```suggestion
- [I built a Rust toolchain for Svelte](https://blog.baseballyama.com/posts/20260721-rsvelte) by Yamagishi Kazutoshi (one of the main Svelte `language-tools` maintainers) walks through `rsvelte`, a Rust reimplementation of the Svelte compiler, formatter, linter, type checker and LSP tested against 12,000+ real components
```
Added a little note to encourage folks to take a look. It really is an incredible feat

- [Three years ago I told you Svelte gave me everything](https://www.reddit.com/r/sveltejs/comments/1uu9sg6/three_years_ago_i_told_you_svelte_gave_me/) by Bishwas Bhandari is a personal retrospective on six years of building with Svelte
- [Building a Dynamic Invoice Form in Svelte 5 with Formisch](https://formisch.dev/blog/dynamic-invoice-form-svelte/) uses Formisch, Valibot and `FieldArray` to build a typed dynamic invoice form
- [Svelte Building Blocks](https://github.com/erayack/svelte-building-blocks) is a free, runnable, level-based course that starts with components and runes and works up through APIs, testing and production

### Libraries, Tools & Components

_Frameworks and Tooling_

- [LayerChart 2.0](https://layerchart.com) shipped a rewrite with a CSS-framework-agnostic API, new components and chart types from Sean Lynch

> - [Mochi 0.8.0](https://mochi.fast) adds an `<Image>` component with resizing, email sending with Svelte components as templates, queues, rate limiting and a built-in captcha

Reviewer: @khromov not sure if you want to highlight other things from Mochi 0.9.0 here too since it's out already

Author: I can mention both

> - [rsvelte](https://github.com/baseballyama/rsvelte) is an in-progress Rust implementation of the full Svelte toolchain from baseballyama

Reviewer: ```suggestion
- [rsvelte](https://baseballyama.github.io/rsvelte/) is a Rust implementation of the Svelte toolchain from Yamagishi Kazutoshi
```
Changed the link to the official site

- [Frizzante v2](https://razshare.github.io/frizzante-docs/) is a Go + Svelte full-stack framework whose latest release adds SSG snapshots
- [SVOCS](https://svocs.dev) is a markdown-first docs and blog generator built on SvelteKit
- [Svelte DocSmith](https://docsmith.geodask.com) is another docs framework for SvelteKit where markdown files become routes and the sidebar builds itself from frontmatter

_UI Components and Visual Effects_

> - [Lily](https://lily-svelte.pages.dev) is a Tailwind v4 component library with a single "quiet by design" visual language and shadcn-style CLI installs ([GitHub](https://github.com/levish0/lily-svelte))

Reviewer: ```suggestion
- [Lily](https://lily-svelte.pages.dev) is a Tailwind v4 component library with a single "quiet by design" visual language and shadcn-style CLI installs
```
There's a GitHub link in the site nav so we can remove this

- [CossUI-Svelte](https://cossui-svelte.com) is an unofficial port of the Coss UI components backfilling components missing from shadcn, Bits UI and Origin UI
- [mindmapcn-svelte](https://mindmapcn-svelte.mind-elixir.com) is a zero-config interactive mind map component styled to fit shadcn-svelte design systems
- [Svelte Swipe To Action](https://svelte-swipe-ashy.vercel.app) is a highly customizable swipe-to-action component
- [Edra v3](https://edra.tsuzat.com) is a rich text editor for SvelteKit that now ships headless and shadcn variants through a CLI and component registry
- [CMS Brew](https://cmsbrew.com/demo) lets clients edit their SvelteKit site through a chat interface that produces plain commits on GitHub

> - [File Viewer for Svelte](https://www.npmjs.com/package/@file-viewer/svelte) is a native Svelte component and action for previewing PDF, Office, CAD and archive files in the browser ([Demo](https://demo.file-viewer.app/?locale=en-US))

Reviewer: ```suggestion
- [File Viewer for Svelte](https://doc.file-viewer.app/guide/ecosystem#svelte) is a native Svelte component and action for previewing PDF, Office, CAD and archive files in the browser ([Demo](https://demo.file-viewer.app/))
```
Replaced link with official site svelte docs

_Developer Tools and Plugins_

- [svelte-vitals](https://oekazuma.github.io/svelte-vitals/) is a static code-health checker for SvelteKit that scores SEO, performance, correctness, security and architecture from source, with inline PR annotations and SARIF upload
- [PerfGraph](https://github.com/Be1zebub/PerfGraph) turns Lighthouse traces into a focused "root cause to impact to fix" report designed to be read by AI agents

> - [SDuX](https://www.sdux-vault.com/docs) is a deterministic pipeline-based state management library that works with Svelte through `@sdux-vault/core`

Reviewer: This might be better under libraries

> - [Vivalence](https://github.com/vivalence/Vivalence) is an AI-harnessed, self-hostable app platform built in Svelte

Reviewer: This might be better under "built with svelte"

> - [Hoikka](https://www.hoikka.dev) is a full-stack SvelteKit ecommerce project that runs on SQLite and deploys to Node or Cloudflare (`pnpx create-hoikka-app` to try it)

Reviewer: This might be better in the built with svelte/frameworks section

> - [svelte-effect-runtime](https://barekey.dev/docs/ser/introduction) lets you `yield*` effectful code directly inside `<script>` and markup

Reviewer: This might be better in the libraries section

> - [PlaySocket](https://therealpaulplay.github.io/PlaySocketJS/) is a multiplayer library that handles optimistic updates and CRDT-based synchronization

Reviewer: This one also sounds like a library
